Registration/RegisterWithAzure.psm1

```diff
@@ -85,6 +104,77 @@ function New-RegistrationLogFile
 
 #region ConnectedScenario
 
+function Get-RegistrationDetailsConnected {
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory = $true)]
+        [PSCredential] $PrivilegedEndpointCredential,
+
+        [Parameter(Mandatory = $true)]
+        [String] $PrivilegedEndpoint,
+
+        [Parameter(Mandatory = $false)]
+        [ValidateNotNullorEmpty()]
+        [PSObject] $AzureContext = (Get-AzureRmContext),
+
+        [Parameter(Mandatory = $true)]
+        [PSCredential] $AzureStackAdminCredential
```

Comment:
if this is done via PEP, we should be able to expose the registration info without needed to ask user for the default provider subscription credential; this flow here won't work with MFA. Instead, we need to check if we already expose registration details in PEP via some like Get-StampInfo or w/e, or if we need to expose a new command that uses our internal service principal identity to query for the resource.

Reply:
This wins comment of the year - thank you!

Reply:
We don't have anything available via PEP yet. I am planning to implement that do. We made this change in registration script to support re-registration on older builds.

Reply:
A few options I guess, but I suppose one simple workaround here for "older" builds would be to make the $AzureStackAdminCredential optional, and trace a warning that it will induce an interactive authentication prompt when the command is executed. And when you call "Login-AzureRMAccount" you can exclude that credential when it is null, and it should induce the interactive login prompt (unless you detect the PEP command to get it without that auth interaction).